### DISCUSSION TITLE
Different conversion for RGB and CMYK pictures

### DISCUSSION THREAD

**Discussion by laah123** [ImageMagick/ImageMagick#2865] [Help]
Hi,
I have a picture folder which contains both rgb and cmyk pictures.
These I want to convert differently. But how can I separate them, using ImageMagick?

If **CMYK** Then
imgMkObj.Convert strFile1img, str**CMYK**profile, strResize, strFlatten, strEmbedProfileAdobeRGB1998, OutputFile

Else
imgMkObj.Convert strFile1img, strRGBprofile, strResize, strFlatten, strEmbedProfileAdobeRGB1998, OutputFile

**Comment by snibgo** [comment]
What version of IM, on what platform? What programming language are you using?

IM can find the colorspace. For example, at the command line:
```
magick in.png -format %[colorspace] info:
```

**Reply by laah123** [reply]
Thanks for your answer.
I am using version 7.0.10-37 on windows 10 and it is a vbs script.

How do I get the content from colorspace info: into a String, so that I can use it in my IF THEN statement?

**Comment by snibgo** [comment]
Sorry, I don't know Visual Basic. The command I showed writes to stdout.

**Reply by laah123** [reply]
strCMYK = imgMkObj.convert(strFileToCheck, "-format", "%r", "info:") - works in VBS
